Firmware/stepper.cpp

```diff
@@ -134,9 +129,7 @@ static uint16_t OCR1A_nominal;
 static uint8_t  step_loops_nominal;
 
 volatile long endstops_trigsteps[3]={0,0,0};
-static volatile bool endstop_x_hit=false;
-static volatile bool endstop_y_hit=false;
-static volatile bool endstop_z_hit=false;
+static volatile uint8_t endstop_hit = 0;
```

Comment:
Please DON'T - these flags are used in time critical code and updating a single bit takes more CPU cycles than writing a single byte.

Reply:
I disagree. Operations on multiple bits at a time are much faster.

This is quicker:
```
endstop_hit = 0;
   3262e:	sts	0x0574, r1	; 0x800574 <_ZL11endstop_hit.lto_priv.323>
###################
 if( endstop_hit) {
   3253a:	lds	r24, 0x0574	; 0x800574 <_ZL11endstop_hit.lto_priv.323>
   3253e:	and	r24, r24
```
Single bit read is Just as fast as before because it had to turn a byte into a bool. Now it uses sbrs instead of tst:
```
   if(endstop_hit & _BV(X_AXIS)) {
   32554:	lds	r24, 0x0574	; 0x800574 <_ZL11endstop_hit.lto_priv.323>
   32558:	sbrs	r24, 0
```

Single bit changes are indeed slower by default. I see where it could be a problem in stepper_check_endstops(), but I just managed to save 192B of flash just by caching volatile values in registers while inside stepper_check_endstops (perfectly safe because of the interrupt part). It's in fact now a lot faster than before. even with caching since I read and write to memory only once and otherwise just do ORI and ANDI on a register

Tested the functionality again using Home, Belt test, Z calibration and MBL. All worked correctly.